docs/proposals/20191030-machine-health-checking.md

```diff
@@ -177,13 +174,32 @@ Out of band:
 
 ### Risks and Mitigations
 
+## Contradictory signal
+
+If a user configures healthchecks such that more than one check applies to
+machines these healthchecks can potentially become a source of contradictory
+information. For instance, one healthcheck could pass for a given machine,
+while another fails. This will result in undefined behavior since the order in
+which the checks are run and the timing of the remediating controller will
+determine which condition is used for remediation decisions.
+
+There are safeguards that can be put in place to prevent this but it is out of
+scope for this proposal. Documentation should be updated to warn users to
+ensure their healthchecks do not overlap.
```

Comment:
Potentially, `clusterctl` could be extended to do validation/conformance in the future

Reply:
That seems weak since it's not a mandatory entrypoint to the system. Validations in clusterctl should only be a better UX for validations that happen in the actual system, similar issue as #3049

Reply:
This validation would be better in a webhook, though I'm not sure how it would work, we can try to validate that MHC selectors don't overlap, but that's complex and not guaranteed to work. So then we have to check that MachineSets don't create machines with labels that overlap in MHC.

This relates back to having multiple remediations at play at once that are independent and don't know about one another. Perhaps we should revisit the hooks proposal as this would prevent this kind of issue I believe